web/src/js/components/Settings/Account/__tests__/AccountComponent.test.js

```diff
@@ -285,4 +293,66 @@ describe('Account component', () => {
     state = wrapper.state()
     expect(state.usernameOpen).toEqual(true)
   })
+
+  it('redirects on clicking the check email form if there is no reauthed parameter', () => {
+    const AccountComponent = require('js/components/Settings/Account/AccountComponent')
+      .default
+    const userData = getMockUserData()
+    delete userData.email
+    const wrapper = mount(<AccountComponent user={userData} />)
+    const buttons = wrapper.find(Button)
+
+    const emailButton = buttons.at(1)
+    emailButton.simulate('click')
+
+    expect(replaceUrl).toHaveBeenCalledWith(loginURL, {
+      next: constructUrl(accountURL, { reauthed: true }),
+      reauth: 'true',
+    })
+  })
+
+  it('redirects on clicking the check email form if there is no reauthed parameter', () => {
+    const AccountComponent = require('js/components/Settings/Account/AccountComponent')
+      .default
+    const userData = getMockUserData()
+    delete userData.email
+    const wrapper = mount(<AccountComponent user={userData} />)
+    const buttons = wrapper.find(Button)
+
+    getUrlParameters.mockReturnValue({ reauthed: true })
+
+    let state = wrapper.state()
+    expect(state.emailOpen).toEqual(false)
+
+    const emailButton = buttons.at(1)
+    emailButton.simulate('click')
+
+    state = wrapper.state()
+    expect(state.emailOpen).toEqual(true)
+
+    expect(replaceUrl).not.toHaveBeenCalled()
+  })
+
+  it('displays email verified popup if the verified url param is provided', () => {
+    const AccountComponent = require('js/components/Settings/Account/AccountComponent')
+      .default
+    const userData = getMockUserData()
+    delete userData.email
+
+    getUrlParameters.mockReturnValue({ verified: true })
+
+    const wrapper = mount(<AccountComponent user={userData} />)
+
+    let state = wrapper.state()
+    expect(state.emailOpen).toEqual(true)
+    expect(state.emailVerified).toEqual(true)
```

Comment:
General feedback (no need to change here): asserting against component state and other internals is not ideal. Rather, focus tests on rendering/output like you do with the assertion below.